test/common/schemaDynamicTest.js

```diff
@@ -93,28 +109,33 @@ SchemaDynamicTest.prototype.standardInvalidArgumentAssertion = function (input,
 		.equal('Expected type ' + expectedType + ' but found type ' + input.expectation);
 };
 
+SchemaDynamicTest.prototype.standardInvalidPropertyAssertion = function (input, expectedType, property, err) {
+	self.standardInvalidArgumentAssertion(input, expectedType, err);
+};
+
 SchemaDynamicTest.prototype.testArgument = function (expectedType, invalidInputs, testedFunction) {
-	var assertion = self.customArgumentAssertion ?  self.customArgumentAssertion.bind(self) : self.standardInvalidArgumentAssertion;
+	var assertion = self.customArgumentAssertion ? self.customArgumentAssertion.bind(self) : self.standardInvalidArgumentAssertion;
 	var test = function (invalidInput, eachCb) {
-		testedFunction(invalidInput.input, function (err) {
+		self.testStyle(testedFunction, invalidInput.input, function (err) {
 			assertion(invalidInput, expectedType, err);
 			eachCb();
 		});
 	};
 	self.carpetTesting(test, invalidInputs, 'should return an error when invoked with %s');
 };
 
-SchemaDynamicTest.prototype.standardInvalidPropertyAssertion = function (input, expectedType, property, err) {
-	self.standardInvalidArgumentAssertion(input, expectedType, err);
-};
 
 SchemaDynamicTest.prototype.testProperty = function (expectedType, invalidInputs, testedFunction, validArgument, property) {
-	var assertion = self.customPropertyAssertion ?  self.customPropertyAssertion.bind(self) : self.standardInvalidPropertyAssertion;
+	var assertion = self.customPropertyAssertion ? self.customPropertyAssertion.bind(self) : self.standardInvalidPropertyAssertion;
 	var test = function (invalidInput, eachCb) {
 		var malformedPart = {};
-		malformedPart[property] = invalidInput.input;
+		if (property.indexOf('.') === -1) {
```

Comment:
not needed I think - setProperty(malformedPart, property, invalidInput.input); is enough. Also change a name for just set please as it is still set lodash function